### PR TITLE
Add openSUSE Leap 15.1 support

### LINKED ISSUE #657: nvidia-docker on opensuse
hello,

I have tried to install nvidia-docker on opensuse leap 42.3 
 I have used centos package:
- nvidia-docker-2.0.2
- nvidia-container-runtime-1.2.1-1
- libnvidia-container_1.0.0
- nvidia-container-runtime-hook-1.2.1-1

After solving many problem in the installation I have finally managed to set all the packages but when I want to test the configuration 

> nvidia-docker run --rm nvidia/cuda nvidia-smi

  I get 
```
container_linux.go:265: starting container process caused "process_linux.go:368: container init caused \"process_linux.go:351: running prestart hook 1 caused \\\"error running hook: exit status 1, stdout: , stderr: exec command: [/usr/local/bin/nvidia-container-cli --load-kmods configure --ldconfig=@/sbin/ldconfig --device=all --compute --utility --require=cuda>=9.0 --pid=6568 /var/lib/docker/btrfs/subvolumes/26850be101a4f175ec4d476f1a056526cd9202da1c4c7dfaaef32b7135ce82fb]\\\\nnvidia-container-cli: initialization error: cuda error: no cuda-capable device is detected\\\\n\\\"\""
docker: Error response from daemon: oci runtime error: container_linux.go:265: starting container process caused "process_linux.go:368: container init caused \"process_linux.go:351: running prestart hook 1 caused \\\"error running hook: exit status 1, stdout: , stderr: exec command: [/usr/local/bin/nvidia-container-cli --load-kmods configure --ldconfig=@/sbin/ldconfig --device=all --compute --utility --require=cuda>=9.0 --pid=6568 /var/lib/docker/btrfs/subvolumes/26850be101a4f175ec4d476f1a056526cd9202da1c4c7dfaaef32b7135ce82fb]\\\\nnvidia-container-cli: initialization error: cuda error: no cuda-capable device is detected\\\\n\\\"\"".
```
I think that the container cannot recognize my gpu and I don't know why ?
 when I run 

> nvidia-smi

I get this output on my host 
```
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 387.26                 Driver Version: 387.26                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 760     On   | 00000000:01:00.0 N/A |                  N/A |
| 34%   27C    P8    N/A /  N/A |    283MiB /  4030MiB |     N/A      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0                    Not Supported                                       |
+-----------------------------------------------------------------------------+
```

Can you please help me?

thank you.

### PR DESCRIPTION
**Pull request by dev-zero**
This requires support for openSUSE in the nvidia-container-runtime, see:

  https://github.com/NVIDIA/nvidia-container-runtime/pull/32

Fixes #657

### CODE REVIEW

**Comment by flx42** [comment]
Thanks for the patches, I didn't have time to test it yet, sorry about that.

**Comment by danielorf** [comment]
As of today, this PR needs the quick fix mentioned by @torhans here:  https://github.com/NVIDIA/nvidia-docker/issues/657#issuecomment-413829563

Otherwise, I have tested successfully on Leap42.3.  Thanks for the work on this one @dev-zero.

**Comment by dev-zero** [comment]
@danielorf merged your PR and rebased the `opensuse-support` branch of my fork of `nvidia-container-runtime` on latest master.

**Comment by danielorf** [comment]
@flx42 - Any idea if/when this will get merged into master?

**Comment by dev-zero** [comment]
btw, I will now update the complete chain for Leap 15, since Nvidia finally provides CUDA for Leap 15.

**Comment by flx42** [comment]
I still didn't have time to test this, but my first reaction would be that we should do only openSUSE Leap 15 for now. Looks like 42.3 didn't get CUDA 10.0:
http://developer.download.nvidia.com/compute/cuda/repos/opensuse423/x86_64/

**Comment by dev-zero** [comment]
Here you go. Better have one supported version of openSUSE than none at all.

File: Makefile [hunk 1]
```diff
@@ -38,6 +38,8 @@ amzn1: $(addsuffix -amzn1, 18.06.1.ce 18.03.1.ce 17.12.1.ce 17.09.1.ce 17.06.2.c
                         -t "nvidia/nvidia-docker2/ubuntu:18.04-docker18.06.1" -f Dockerfile.ubuntu .
 	$(DOCKER) run --rm -v $(DIST_DIR)/ubuntu18.04:/dist:Z "nvidia/nvidia-docker2/ubuntu:18.04-docker18.06.1"
 
+opensuse_leap15.0: $(addsuffix -opensuse_leap15.0, 17.09.1_ce)
```
[the remark this answers is not in the document]
Comment (dev-zero) [reply Makefile]: On LEAP 15.1 it seems only 18.06.1 and 18.06.6 are available, I've added both of them

**Comment by dirkmueller** [comment]
Is there any progress here? I still see no merges..

File: README.md [hunk 2]
```diff
@@ -88,6 +88,22 @@ sudo yum install -y nvidia-container-runtime-hook
 docker run --rm nvidia/cuda:9.0-base nvidia-smi
 ```
 
+#### openSUSE Leap 15.1 (docker-ce)
```
Comment (RenaudWasTaken) [inline README.md]: Can you rebase over the current readme?

Comment (dev-zero) [reply README.md]: done

File: Dockerfile.opensuse-leap [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+COPY rpm .
+
+CMD rpmbuild --clean -bb \
+             -D "_topdir $PWD" \
+             -D "version $VERSION" \
+             -D "release $RELEASE" \
+             -D "docker_version $DOCKER_VERSION" \
+             -D "runtime_version $RUNTIME_VERSION" \
+             SPECS/nvidia-docker2.spec && \
+    mv RPMS/noarch/*.rpm /dist
```
Comment (Ethyling) [inline Dockerfile.opensuse-leap]: Can you add this line before the `rpmbuild`: `RUN sed -i "s;@VERSION@;${PKG_VERS};" $DIST_DIR/nvidia-docker`. We need this to fix the nvidia-docker version

**Comment by RenaudWasTaken** [comment]
@dev-zero can you address @Ethyling 's comment :) ?
Thanks!

File: Dockerfile.opensuse-leap (hunk 3 above)
Comment (dev-zero) [reply Dockerfile.opensuse-leap]: done

**Comment by dev-zero** [comment]
@RenaudWasTaken done, thanks for the reminder

**Comment by Ethyling** [comment]
LGTM.
This repo is now mirrored from https://gitlab.com/nvidia/container-toolkit/nvidia-container-runtime. Could you do your PR there? Thank you!

File: Dockerfile.centos [hunk 4]
```diff
@@ -15,7 +15,7 @@ ARG DOCKER_VERSION
 ENV VERSION $PKG_VERS
 ENV RELEASE $PKG_REV
 ENV DOCKER_VERSION $DOCKER_VERSION
-ENV DOCKER_VERSION $DOCKER_VERSION
+ENV RUNTIME_VERSION $RUNTIME_VERSION
```
Comment (RenaudWasTaken) [inline Dockerfile.centos]: Thanks for spotting this one!

**Comment by RenaudWasTaken** [comment]
Merged in ee1c56f

I did this manually as we moved our code to gitlab.com/NVIDIA
These are still your changes that were merged :)

Sorry this took so long, I expect to have time to release the packages next week.

**Pull request closed without merging**